Blocks/HTTP/request.h

```diff
@@ -44,45 +45,59 @@ constexpr static auto HasRespondViaHTTP(int)
   return true;
 }
 
-struct URLPathParams {
+struct URLPathArgs {
  public:
-  enum class Count : uint16_t { None = 0u, One = 1u, Two = 2u, Any = 0xFFFF };
+  using CountMaskUnderlyingType = uint16_t;
+  enum class CountMask : CountMaskUnderlyingType {
+    None = (1 << 0),
+    One = (1 << 1),
+    Two = (1 << 2),
+    Three = (1 << 3),
+    Any = static_cast<CountMaskUnderlyingType>(~0)
+  };
 
-  const std::string& operator[](size_t index) const { return params_accessor_.at(index); }
+  const std::string& operator[](size_t index) const { return args_.at(args_.size() - 1 - index); }
 
-  using T_ITERATOR = std::vector<std::reference_wrapper<std::string>>::const_reverse_iterator;
-  T_ITERATOR begin() const { return params_accessor_.crbegin(); }
-  T_ITERATOR end() const { return params_accessor_.crend(); }
+  using T_ITERATOR = std::vector<std::string>::const_reverse_iterator;
+  T_ITERATOR begin() const { return args_.crbegin(); }
+  T_ITERATOR end() const { return args_.crend(); }
 
-  bool empty() const { return params_accessor_.empty(); }
-  bool size() const { return params_accessor_.size(); }
+  bool empty() const { return args_.empty(); }
+  bool size() const { return args_.size(); }
 
-  void add(const std::string& value) {
-    param_values_.push(value);
-    params_accessor_.emplace_back(param_values_.top());
-  }
+  void add(const std::string& arg) { args_.push_back(arg); }
 
   std::string base_path;
 
- private:
-  std::stack<std::string> param_values_;
-  std::vector<std::reference_wrapper<std::string>> params_accessor_;
+// WTF! WTF!! WTF!!!!!!!1111
+// URLPathArgs.size() and URLPathArgs.args_.size() give different results when running the test.
```

Comment:
Yes, I'm surprised none of our compilers gave a warning. Microsoft's one
would surely have.